<a href="https://colab.research.google.com/github/sparshmittal/Regex/blob/main/Regex_Sample_Data_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import re

In [1]:
name = input('Enter file name')

Enter file nameSample Data 1


In [ ]:
file = open("/content/"+name+".txt", "r")
data = file.read()
print(file.read())

The Samsung Galaxy A53 5G price is $449.99 / £399 / AU$699. That gets you the 6GB of RAM with 128GB of storage model - and it’s the only variant of the handset Samsung is offering, in the UK and Australia at least.

The price sees the Galaxy A53 rub shoulders with other 5G-toting mid-rangers, including the Mi 11 Lite, OnePlus Nord 2, Moto G200 and the new iPhone SE 2022. 

The Samsung Galaxy A53 5G release date is set for 04-01-2022 in the UK and Australia, with pre-orders opening today, 03-17-2022.


In [ ]:
data

'The Samsung Galaxy A53 5G price is $449.99 / £399 / AU$699. That gets you the 6GB of RAM with 128GB of storage model - and it’s the only variant of the handset Samsung is offering, in the UK and Australia at least.\n\nThe price sees the Galaxy A53 rub shoulders with other 5G-toting mid-rangers, including the Mi 11 Lite, OnePlus Nord 2, Moto G200 and the new iPhone SE 2022. \n\nThe Samsung Galaxy A53 5G release date is set for 04-01-2022 in the UK and Australia, with pre-orders opening today, 03-17-2022.'

In [ ]:
data = data.replace('\n',' ')

In [ ]:
print(data)

The Samsung Galaxy A53 5G price is $449.99 / £399 / AU$699. That gets you the 6GB of RAM with 128GB of storage model - and it’s the only variant of the handset Samsung is offering, in the UK and Australia at least.  The price sees the Galaxy A53 rub shoulders with other 5G-toting mid-rangers, including the Mi 11 Lite, OnePlus Nord 2, Moto G200 and the new iPhone SE 2022.   The Samsung Galaxy A53 5G release date is set for 04-01-2022 in the UK and Australia, with pre-orders opening today, 03-17-2022.


**1. Print all currencies in text**

Our text has following currencies - \$449.99 / £399 / AU\$699

(A?U?[\$£]) <--- checks the currency symbol. We can add more currency symbols in the square brackets. 

[0-9]+      <--- checks digit after currency

(\.[0-9]+)?)<--- checks digits after decimal if any




In [ ]:
currency = re.findall(r'(A?U?[\$£]([0-9]+(\.[0-9]+)?))', data)
for curr in currency:
  print(curr[0])

$449.99
£399
AU$699


**2. Print all date times in the text**

(0?[1-9]|1[0-2])   <-- To check month->mm because only 0/1 as fist digit of the month and only 10,11,12 for first digit 1.

(0[1-9]|1[0-9]|2[0-9]|3[0-1])   <-- To check date->dd because only 0/1/2/3 as fist digit of the date with diff 2nd digits.Also range of date is 1-31.

\d{4}   <--- To check the year->yyyy beacuse we can assume any 4 digits can be possible in year

we can arrange these to get desired format - dd/mm/yyyy, dd/mm/yy, mm/dd/yyyy, mm/dd/yy

In [ ]:
dates = re.findall(r'((0?[1-9]|1[0-2])(-)((0[1-9]|1[0-9]|2[0-9]|3[0-1]))(-)(\d{4}))',data)
for date in dates:
  print(date[0])

04-01-2022
03-17-2022


**3. Print all cardinilities and orders-**

For 1st/2nd/3rd etc -  ([0-9]+) for number and (st|nd|rd|th) for the suffix

For first/second/nineteenth etc if we use ([a-zA-Z]+)(st|nd|rd|th), it also shows other words like with/last etc. And then there are different rules for words like twenty - twentieth, twenty one - twenty first, twelve - twelfth so we cant match them on word names of numbers either.

In [ ]:
#no cardinality/order in this data
orders = re.findall(r'(([0-9]+)(st|nd|rd|th))',data)
for order in orders:
  print(order[0])

**4. Print all 4 letter words that begin with vowels**

\s at front and back to check for spaces

[aeiouAEIOU] to check if first letter is vowel

[a-zA-Z]{3}   for remaining 3 letters

In [ ]:
words = re.findall(r'\s[aeiouAEIOU][a-zA-Z]{3}\s',data)
for word in words:
  print(word)

 only 
